src/coaster_gui.cpp

```diff
@@ -212,6 +350,101 @@ void CoasterInstanceWindow::SetRadioChecked(WidgetNumber radio, bool checked)
 	this->MarkWidgetDirty(radio);
 }
 
+void CoasterInstanceWindow::OnChange(const ChangeCode code, const uint32 parameter)
+{
+	switch (code) {
+		case CHG_DISPLAY_OLD:
+			this->MarkDirty();
+			break;
+		case CHG_DROPDOWN_RESULT:
+			switch ((parameter >> 16) & 0xFF) {
+				case CIW_CHOOSE_ENTRANCE:
+					this->ci->SetEntranceType(parameter & 0xFF);
+					this->UpdateRecolourButtons();
+					break;
+				case CIW_CHOOSE_EXIT:
+					this->ci->SetExitType(parameter & 0xFF);
+					this->UpdateRecolourButtons();
+					break;
+				default:
+					break;
+			}
+			break;
+		default:
+			break;
+	}
+}
+
+void CoasterInstanceWindow::SelectorMouseMoveEvent(Viewport *vp, const Point16 &pos)
+{
+	const Point32 world_pos = vp->ComputeHorizontalTranslation(vp->rect.width / 2 - pos.x, vp->rect.height / 2 - pos.y);
+	int dx, dy;
+	switch (vp->orientation) {
+		case VOR_NORTH: dx =  1; dy =  1; break;
+		case VOR_WEST:  dx = -1; dy =  1; break;
+		case VOR_SOUTH: dx = -1; dy = -1; break;
+		case VOR_EAST:  dx =  1; dy = -1; break;
+		default: NOT_REACHED();
```

Comment:
This looks like it's common code, other occurrences are at `gentle_thrill_ride_gui.cpp` line 365, and `ride_build.cpp` line 291.  Perhaps move it to a common function?

Reply:
Perhaps similar to the short arrays in tile_func.cpp at the start, like `_corners_at_edge` (and declaration in `tile.h`). However, being a computation in view orientation, some `orientation_func.cpp` would be better (We only have `orientations.h` currently.)